Source/DafnyLanguageServer/Workspace/DafnyDocument.cs

```diff
@@ -32,6 +35,29 @@ public record DafnyDocument(
     /// </summary>
     public string? SerializedCounterExamples { get; init; }
 
+
+    /// <summary>
+    /// True is the resolution succeeded, false if resolution failed
+    /// <c>null</c> If the verification did not start (e.g. because of resolution errors)
+    /// </summary>
+    public bool? ResolutionSucceeded { get; set; } = null;
```

Comment:
`ResolutionSuceeded` was never using "set", only initialization, so I replaced `set` by `init` to make it immutable.
I removed `VerificationPass` because I could make it an argument to the methods that were using it.
However, for `VerificationTree`, I don't think it's appropriate. I tried to do my best so that VerificationTree scales with the file, so mutability is the best option I had. Moreover, this [follow-up PR](https://github.com/dafny-lang/dafny/pull/1946/files) relies on accessing the verification tree in real-time, and it only has access to the current `DafnyDocument`.
Mutability can be a burden, but in this case, I think it's appropriate to keep it.

Reply:
We can keep it like this and see for now, as long as we try to keep DafnyDocument immutable :-)